pkg/analysis/requiredfields/testdata/src/b/a.go

```diff
@@ -1,39 +1,16 @@
-package a
+package c
 
-type A struct {
-	// optional field should not be picked up.
-	// +optional
-	OptionalField *string `json:"optionalField,omitempty"`
-
-	// requiredCorrectField should not be picked up.
-	// +required
-	RequiredCorrectField string `json:"requiredCorrectField"`
-
-	// requiredOmitEmptyField field should be picked up.
+type Foo struct {
 	// +required
-	RequiredOmitEmptyField string `json:"requiredOmitEmptyField,omitempty"` // want "field RequiredOmitEmptyField is marked as required, but has the omitempty tag"
+	RequiredStructWithRequiredFieldWithoutOmitZero StructWithRequiredField `json:"dataplane"` // want "field Foo.RequiredStructWithRequiredFieldWithoutOmitZero does not allow the zero value. It must have the omitzero tag."
```

Comment:
```suggestion
	RequiredStructWithRequiredFieldWithoutOmitZero StructWithRequiredField `json:"requiredStructWithRequiredFieldWithoutOmitZero"` // want "field Foo.RequiredStructWithRequiredFieldWithoutOmitZero does not allow the zero value. It must have the omitzero tag."
```

Why do you add another name as json tag ? Is it implicit ?